### ISSUE TITLE
Cross-validation Finished_2.00%Error in mult.store[, 2] : subscript out of bounds

### ISSUE THREAD

**Issue by Markjiancheng** [YinLiLin/R-KAMBLUP#12]
Hi, hope you are doing well.
I am trying KAML for some data but I always get this error:
Cross-validation Finished_2.00%Error in mult.store[, 2] : subscript out of bounds

And this for a different data: 
 Pick up pseudo QTNs at LD threshold(0.3) as following:
Error in if (sum(p_cor > cor.threshold) == 0) bin.qtn[[g]] <- c(bin.qtn[[g]],  :
  missing value where TRUE/FALSE needed

Would you please help with these issues? Thanks!

**Comment by YinLiLin** [comment]
Hi Mark,

Thanks for trying KAML, we haven't caught this error by now using the data in hand, may i please ask for a small dataset which can repeat the error as you proposed? i am pleased to debug the script of KAML for this issue, and feed back to you as soon as possible.

Best,
Lilin

**Comment by Markjiancheng** [comment]
Hi YinLiLin,

Thanks for your prompt reply, I actually tried the horse and maize data that you used in your paper, but I am not sure why both of them did not work.
Attached is the exact code and data I used, please see attached
[GDD.zip](https://github.com/YinLiLin/KAML/files/7445587/GDD.zip)
[Horse.zip](https://github.com/YinLiLin/KAML/files/7445588/Horse.zip)
!

**Comment by YinLiLin** [comment]
Hi, I just tried both of the data attached, it works fine on my computer, the LOG information can be found below:
[horse.log](https://github.com/YinLiLin/KAML/files/7450120/horse.log)
[GDD.log](https://github.com/YinLiLin/KAML/files/7450122/GDD.log)
Did you use the latest version of KAML that are installed from Github?

**Comment by Markjiancheng** [comment]
I think so. I just installed according to the instruction recently.

**Comment by YinLiLin** [comment]
Thank you for pointing out this issue, we have located the problem, and the latest script on GitHub can be run successfully on different version of R, please try again.

**Comment by Markjiancheng** [comment]
Thanks for checking it out, however, the problem still exist:
Error in if (sum(p_cor > cor.threshold) == 0) bin.qtn[[g]] <- c(bin.qtn[[g]], :
missing value where TRUE/FALSE needed

Moreover, it looks like computing grm took me longer than before, is it because that you updated the package?

**Comment by YinLiLin** [comment]
Yes, we have changed the computing strategy for GRM construction, it take less memory than previous version. Can the updated version run "horse" and "GDD" data successfully on your computer? Or it just failed for your own data? if in this case, could you mind sending me a small data? i need to repeat this issue to get it fixed.

**Comment by Markjiancheng** [comment]
No, it failed on the GDD data.

**Comment by YinLiLin** [comment]
May i know the version of R you used and the operating system (linux, windows, mac) of your computer?

**Comment by Markjiancheng** [comment]
It is R 4.1.1 in Linux.

**Comment by YinLiLin** [comment]
I just ran KAML at the same environment as yours (R 4.1.1, Linux), it can still work without problem, please see the log file below:
[KAML_R-4.1.1_run.log](https://github.com/YinLiLin/KAML/files/7511062/KAML_R-4.1.1_run.log)
Sorry that it seems there may be potential problem which would be encountered at specify systematical environment, so I recommend to have a try at another computer.
